code/game/objects/structures/mirror.dm

```diff
@@ -36,44 +73,149 @@ MAPPING_DIRECTIONAL_HELPERS(/obj/structure/mirror, 28)
+		return TRUE //no tele-grooming (if nonmagical)
+
+	return what_is_your_wish_masta(user)
+
+/obj/structure/mirror/proc/what_is_your_wish_masta(mob/living/carbon/human/user)
+	var/pick = show_radial_menu(user, src, mirror_options, user, radius = 36, require_near = TRUE)
+	if(!pick)
+		return TRUE //get out
+
+	switch(pick)
+		if(CHANGE_HAIR)
+			change_hair(user)
+		if(CHANGE_BEARD)
+			change_beard(user)
+		if(CHANGE_RACE)
+			change_race(user)
+		if(CHANGE_SEX) // sex: yes
+			change_sex(user)
+		if(CHANGE_NAME)
+			change_name(user)
+		if(CHANGE_EYES)
+			change_eyes(user)
+
+	return what_is_your_wish_masta(user)
+
+/obj/structure/mirror/proc/change_beard(mob/living/carbon/human/beardresser)
+	if(beardresser.physique != FEMALE && !magical_mirror)
+		var/new_style = tgui_input_list(beardresser, "Select a facial hairstyle", "Grooming", GLOB.facial_hairstyles_list)
 		if(isnull(new_style))
 			return TRUE
-		if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH))
-			return TRUE //no tele-grooming
-		if(HAS_TRAIT(hairdresser, TRAIT_SHAVED))
-			to_chat(hairdresser, span_notice("If only growing back facial hair were that easy for you..."))
+		if(HAS_TRAIT(beardresser, TRAIT_SHAVED))
+			to_chat(beardresser, span_notice("If only growing back facial hair were that easy for you... The reminder makes you feel terrible."))
+			beardresser.add_mood_event("bald_hair_day", /datum/mood_event/bald)
 			return TRUE
-		hairdresser.set_facial_hairstyle(new_style, update = TRUE)
+		beardresser.set_facial_hairstyle(new_style, update = TRUE)
 	else
-		hairdresser.set_facial_hairstyle("Shaved", update = TRUE)
+		if(beardresser.facial_hairstyle == "Shaved")
+			to_chat(beardresser, span_notice("You look towards your facial hair for a make-up and realize you don't have any."))
+			return
+		beardresser.set_facial_hairstyle("Shaved", update = TRUE)
 
-	//handle normal hair
-	var/new_style = tgui_input_list(user, "Select a hairstyle", "Grooming", GLOB.hairstyles_list)
+/obj/structure/mirror/proc/change_hair(mob/living/carbon/human/hairdresser)
+	var/new_style = tgui_input_list(hairdresser, "Select a hairstyle", "Grooming", GLOB.hairstyles_list)
 	if(isnull(new_style))
 		return TRUE
-	if(!user.can_perform_action(src, FORBID_TELEKINESIS_REACH))
-		return TRUE //no tele-grooming
 	if(HAS_TRAIT(hairdresser, TRAIT_BALD))
-		to_chat(hairdresser, span_notice("If only growing back hair were that easy for you..."))
+		to_chat(hairdresser, span_notice("If only growing back hair were that easy for you... The reminder makes you feel terrible."))
+		hairdresser.add_mood_event("bald_hair_day", /datum/mood_event/bald)
 		return TRUE
 
 	hairdresser.set_hairstyle(new_style, update = TRUE)
 
-/obj/structure/mirror/examine_status(mob/user)
+/obj/structure/mirror/proc/change_name(mob/living/carbon/human/user)
+	var/newname = sanitize_name(tgui_input_text(user, "Who are we again?", "Name change", user.name, MAX_NAME_LEN), allow_numbers = TRUE) //It's magic so whatever.
```

Comment:
this should be `snake_case` so make this `var/new_name`

Reply:
also it's crazy that allow numbers was set to true. what were old coders cooking. we can keep it but that's really strange

Reply:
idk what if the wiz wanted to larp as an android technomancer.